test/Microsoft.Azure.WebJobs.Host.UnitTests/JobHostMetadataProviderTests.cs

```diff
@@ -236,6 +235,58 @@ public void Initialize(ExtensionConfigContext context)
             }
         }
 
+
+        [Fact]
+        public void DefaultTypeForTrigger()
+        {
+            var ext = new JArrayTriggerExtension();
+            var prog = new FakeTypeLocator();
+            JobHostConfiguration config = TestHelpers.NewConfig(prog, ext);
+            var host = new JobHost(config);
+            IJobHostMetadataProvider metadataProvider = host.CreateMetadataProvider();
+
+            var attr = new Test9Attribute(null);
+            var type = metadataProvider.GetDefaultType(attr, FileAccess.Read, null);
+
+            Assert.Equal(typeof(JArray), type);
+        }
+
+        public class JArrayTriggerExtension : IExtensionConfigProvider
+        {
+            public void Initialize(ExtensionConfigContext context)
+            {
+                var rule = context.AddBindingRule<Test9Attribute>();
+                rule.BindToTrigger<string>();
+                rule.AddConverter<string, JArray>(input => (JArray)null);
+            }
+        }
+
+        [Fact]
+        public void DefaultTypeForOpenTypeTrigger()
+        {
+            var ext = new OpenTypeTriggerExtension();
+            var prog = new FakeTypeLocator();
+            JobHostConfiguration config = TestHelpers.NewConfig(prog, ext);
+            var host = new JobHost(config);
+            IJobHostMetadataProvider metadataProvider = host.CreateMetadataProvider();
+
+            var attr = new Test9Attribute(null);
+            var type = metadataProvider.GetDefaultType(attr, FileAccess.Write, null);
+
+            // The trigger handles Open type, which means it will first pull byte[]. 
+            Assert.Equal(typeof(byte[]), type);
+        }
+
+        public class OpenTypeTriggerExtension : IExtensionConfigProvider
+        {
+            public void Initialize(ExtensionConfigContext context)
+            {
+                var rule = context.AddBindingRule<Test9Attribute>();
+                rule.BindToTrigger<string>();
+                rule.AddOpenConverter<string, OpenType>((a, b, c) => null);
```

Comment:
I wanted to see if this worked for OpenTypes; looks like it does. Am I registering this the right way?